### PR TITLE
[WB-3442] Ground truth overriding annotations

### PR DESCRIPTION
**Pull request by jsbroks**
### Cause

When saving annotations to JSON file the same id(/filename) was being used. When not in an array no id_ is set, so the annotations hash is used instead.

### Fix

Set the id using the image index and annotation index. Previously it was just using the image index. Not really sure if this is the best way...

I added a `None` check so that if no id is passed in, it would default to using the annotations hash. This was done to keep the same behaviour as previously.

I also tried setting the `id_` to `None` for all annotations but this failed when passing in an array with 2 of the same images since they both have the same media hash.

### Testing

1. wandb.log tests, test all cases with bounding boxes and masks
```python
wandb.log({'data': img})
wandb.log({'data': [img]})
wandb.log({'data': [img, img]})
wandb.log({'data': [img, img2]})
```

2. Ran `tox`

### CODE REVIEW

**Review by shawnlewis — APPROVED** [review]
This seems correct to me. Nice work. Let's wait for Jeff for final approval before merge.

**Comment by raubitsj** [comment]
circle didnt run, not sure why,  I have noticed this happening on forked PRs.  I would like to get that running before merging.

**Pull request merged**
